submit/controllers/process.py

```diff
@@ -0,0 +1,142 @@
+
+from arxiv import status
+from arxiv.base import logging, alerts
+from arxiv.forms import csrf
+from arxiv.users.domain import Session
+from ..services import compiler
+
+logger = logging.getLogger(__name__)
+
+Response = Tuple[Dict[str, Any], int, Dict[str, Any]]  # pylint: disable=C0103
+
+def file_process(method: str, session: Session, submission_id: int, token: str) -> Response:
+    """
+    Process the file compilation project.
+
+    Parameters
+    ----------
+    method : str
+        ``GET`` or ``POST``
+    session : :class:`Session`
+        The authenticated session for the request.
+    submission_id : int
+        The identifier of the submission for which the upload is being made.
+    token : str
+        The original (encrypted) auth token on the request. Used to perform
+        subrequests to the file management service.
+
+    Returns
+    -------
+    dict
+        Response data, to render in template.
+    int
+        HTTP status code. This should be ``200`` or ``303``, unless something
+        goes wrong.
+    dict
+        Extra headers to add/update on the response. This should include
+        the `Location` header for use in the 303 redirect response, if
+        applicable.
+    """
+    if method == "GET":
+        return compile_status(session, submission_id, token)
+    #elif method == "POST":
+    #    return compile(seesion, submission_id, token)
+    else:
+        return {}, status.HTTP_400_BAD_REQUEST, {}
+
+def compile_status(session: Session, submission_id: int, token: str) -> Response:
+    """
+    Returns the status of a compilation.
+
+    Parameters
+    ----------
+    session : :class:`Session`
+        The authenticated session for the request.
+    submission_id : int
+        The identifier of the submission for which the upload is being made.
+    token : str
+        The original (encrypted) auth token on the request. Used to perform
+        subrequests to the file management service.
+
+    Returns
+    -------
+    dict
+        Response data, to render in template.
+    int
+        HTTP status code. This should be ``200`` or ``303``, unless something
+        goes wrong.
+    dict
+        Extra headers to add/update on the response. This should include
+        the `Location` header for use in the 303 redirect response, if
+        applicable.
+    """
+    compiler.set_auth_token(token)
+    try:
+        compiler.request_compilation(submission_id)
+    except compiler.BadRequest as e:
+        logger.debug(f'Bad request to compiler for {submission_id}')
+        return {'status': 'failed'}, status.HTTP_400_BAD_REQUEST, {}
+
+    redirect = url_for('ui.file_process', submission_id=submission_id)
+    return {}, status.HTTP_303_SEE_OTHER, {'Location': redirect}
+
+def compile(session: Session, submission_id: int, token: str) -> Response:
+    redirect = url_for('ui.file_process', submission_id=submission_id)
+    return {}, status.HTTP_303_SEE_OTHER, {'Location': redirect}
```

Comment:
Nit: add return at EOF as GitHub is hinting

Reply:
Fixed